src/pages/manage/productions/create.js

```diff
@@ -1,28 +1,314 @@
+        disabled={shouldDisableLinkButton}
+        onClick={onClickLink}
+      >
+        {t('productions.create.link')}
+      </Button>
+      <Button variant={ButtonVariants.DANGER} onClick={onClickSkip}>
+        {t('productions.create.skip')}
+      </Button>
+    </Inline>
+  );
+};
+
+LinkAndSkipButtons.propTypes = {
+  shouldDisableLinkButton: PropTypes.bool,
+  onClickLink: PropTypes.func,
+  onClickSkip: PropTypes.func,
+};
+
+const ProductionStatus = {
+  MISSING: 'missing',
+  EXISTING_SELECTED: 'existingSelected',
+  NEW: 'new',
+};
 
 const Create = () => {
-  return <Title>Productions - create</Title>;
-};
-
-export const getServerSideProps = async ({ req, query }) => {
-  const { cookies } = new Cookies(req?.headers?.cookie);
-  const isUnAuthorized = !cookies.token && !query?.jwt;
-
-  if (isUnAuthorized) {
-    return {
-      redirect: {
-        destination: '/login',
-        permanent: false,
-      },
-    };
-  }
-
-  return {
-    props: {
-      cookies,
-    },
+  const { t, i18n } = useTranslation();
+  const queryClient = useQueryClient();
+  const [searchInput, setSearchInput] = useState('');
+  const [selectedProductionId, setSelectedProductionId] = useState('');
+
+  const {
+    data: suggestedEvents,
+    status: suggestedEventsStatus,
+    refetch: refetchSuggestedEvents,
+  } = useGetSuggestedEvents();
+  const { data: suggestedProductionsData } = useGetProductions({
+    name: searchInput,
+    limit: 10,
+  });
+
+  const handleSuccess = async () => {
+    setSelectedProductionId('');
+    setSearchInput('');
+    await refetchSuggestedEvents();
+    await queryClient.refetchQueries(['productions']);
   };
+
+  const { mutate: skipSuggestedEvents } = useSkipSuggestedEvents({
+    onSuccess: handleSuccess,
+  });
+
+  const { mutate: createProductionWithEvents } = useCreateWithEvents({
+    onSuccess: handleSuccess,
+  });
+
+  const { mutate: mergeProductions } = useMergeProductions({
+    onSuccess: handleSuccess,
+  });
+
+  const { mutate: addEventsByIds } = useAddEventsByIds({
+    onSuccess: handleSuccess,
+  });
+
+  const suggestedProductions = suggestedProductionsData?.member ?? [];
+  const events = suggestedEvents?.events ?? [];
+  const similarity = suggestedEvents?.similarity ?? 0;
+
+  const productions = useMemo(
+    () =>
+      events
+        .map((event) => event?.production)
+        .filter((production) => !!production),
+    [events],
+  );
+
+  useEffect(() => {
+    if (productions.length === 1) {
+      setSelectedProductionId(productions[0].id);
+    }
+  }, [productions]);
+
+  const status = useMemo(() => {
+    if (selectedProductionId) {
+      return ProductionStatus.EXISTING_SELECTED;
+    }
+    if (searchInput) {
+      return ProductionStatus.NEW;
+    }
+    return ProductionStatus.MISSING;
+  }, [selectedProductionId, searchInput]);
+
+  const handleInputSearch = (searchTerm) => {
+    debounce(() => setSearchInput(searchTerm.toString().trim()), 275)();
```

Comment:
Do we still need this with the debouncing in the typeahead itself?

Reply:
The internal debouncing was causing problems, so I moved it to be our own